framework/platform/bpl/include/bpl/bpl_db.h

```diff
@@ -0,0 +1,87 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _BPL_DB_H_
+#define _BPL_DB_H_
+
+#include "bpl.h"
+#include "bpl_err.h"
+#include <stdint.h>
+#include <string>
+#include <unordered_map>
+
+namespace beerocks {
+namespace bpl {
+
+/****************************************************************************/
+/******************************* Definitions ********************************/
+/****************************************************************************/
+
+/****************************************************************************/
+/******************************* Structures *********************************/
+/****************************************************************************/
+
+/****************************************************************************/
+/******************************** Functions *********************************/
+/****************************************************************************/
+
+/**
+ * @brief Add new entry with initial value.
+ * 
+ * @param[in] entry_type type of the requested entry.
+ * @param[in] entry_name name of the requested entry.
+ * @param[in] values initial values of the entry.
+ * @return true on success, false otherwise
+ */
+bool db_add_entry(const std::string &entry_type, const std::string &entry_name,
+                  const std::unordered_map<std::string, std::string> &values);
+/**
+ * @brief Set values in entry, append attributes and override existing.
+ * 
+ * @param[in] entry_type type of the requested entry.
+ * @param[in] entry_name name of the requested entry.
+ * @param[in] values unordered_map containing the values the requested entry, to be set.
+ * @return true on success, false otherwise.
+ */
+bool db_set_entry_params(const std::string &entry_type, const std::string &entry_name,
```

Comment:
What is the `entry_type` used for? If the `entry_name` is unique then no other value is required for disambiguation to look up the entry.
This comment applies to the rest of functions below too.

Reply:
see my comment https://github.com/prplfoundation/prplMesh/pull/1460#issuecomment-651028711